_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1066,6 +1075,95 @@ private function _process_onboarding( $data ) {
 			: join( ', ', $error );
 	}
 
+	/**
+	 * Retrieve the first active sidebar.
+	 *
+	 * @return array|WP_Error First active sidebar, error if none exists.
+	*/
+	static function get_first_sidebar() {
+		$active_sidebars = get_option( 'sidebars_widgets', array() );
+		unset( $active_sidebars[ 'wp_inactive_widgets' ], $active_sidebars[ 'array_version' ] );
+
+		if ( empty( $active_sidebars ) ) {
+			return new WP_Error( 'invalid_data', 'Invalid data.', 400 );
+		}
+		$active_sidebars_keys = array_keys( $active_sidebars );
+		return array_shift( $active_sidebars_keys );
+	}
+
+	/**
+	 * Add or update Business Address widget.
+	 *
+	 * @param array $business_address_content Value corresponding to the Business
+	 * Address as received from the settings endpoint.
+	 *
+	 * @return WP_Error|true True when data has been saved correctly,
+	 * error otherwise.
+	*/
+	static function handle_business_address( $business_address_content ) {
+		$first_sidebar = self::get_first_sidebar();
+
+		$widgets_module_active = Jetpack::is_module_active( 'widgets' );
+		if ( ! $widgets_module_active ) {
+			$widgets_module_active = Jetpack::activate_module( 'widgets', false, false );
+		}
+		if ( ! $widgets_module_active ) {
+			return new WP_Error( 'invalid_data', 'Failed to activate module.', 400 );
+		}
+
+		if ( $first_sidebar ) {
+			$title = $business_address_content['name'];
```

Comment:
We need to sanitize this one, because right now we'll insert them plain, and that will allow for scripts and other malicious contents. For sanitization you can use `sanitize_text_field()`.